benchmark/Main.hs

```diff
@@ -20,28 +20,49 @@ import Gauge.Main
 --------------------------------------------------------------------------------
 
 {-# INLINE large_bible_txt #-}
-large_bible_txt :: String -> String
-large_bible_txt base = base ++ "large/bible.txt"
+large_bible_txt :: String
+large_bible_txt = "large/bible.txt"
 
 {-# INLINE large_world192_txt #-}
-large_world192_txt :: String -> String
-large_world192_txt base = base ++ "large/world192.txt"
+large_world192_txt :: String
+large_world192_txt = "large/world192.txt"
 
 {-# INLINE cantrbry_alice29_txt #-}
-cantrbry_alice29_txt :: String -> String
-cantrbry_alice29_txt base = base ++ "cantrbry/alice29.txt"
+cantrbry_alice29_txt :: String
+cantrbry_alice29_txt = "cantrbry/alice29.txt"
 
-{-# INLINE cantrbry_kennedy_xls #-}
-cantrbry_kennedy_xls :: String -> String
-cantrbry_kennedy_xls base = base ++ "cantrbry/kennedy.xls"
-
-{-# INLINE artificl_random_txt #-}
-artificl_random_txt :: String -> String
-artificl_random_txt base = base ++ "artificl/random.txt"
+--------------------------------------------------------------------------------
+-- Bootstrapping
+--------------------------------------------------------------------------------
 
-{-# INLINE artificl_aaa_txt #-}
-artificl_aaa_txt :: String -> String
-artificl_aaa_txt base = base ++ "artificl/aaa.txt"
+normalizedName :: String -> String
+normalizedName x = x ++ ".normalized"
+
+bigCompressedName :: String -> String
+bigCompressedName x = x ++ ".normalized.compressed.big"
+
+smallCompressedName :: String -> String
+smallCompressedName x = x ++ ".normalized.compressed.small"
+
+bootstrap :: String -> IO ()
+bootstrap file = do
+    base <- getCurrentDirectory
+    let fp = base ++ "/corpora/" ++ file
+        normalizedFp = normalizedName fp
+        compressedFpBig = bigCompressedName fp
+        compressedFpSmall = smallCompressedName fp
+    fileExists <- doesFileExist normalizedFp
+    unless fileExists $ do
+        lst <-
+            Stream.unfold File.readChunksWithBufferOf (64 * 1024, fp)
+                & Stream.toList
+        -- 160 * 64 * 1024 = 10MB
+        let lst1 = take 160 $ cycle lst
```

Comment:
The arrays may not be 64K. The read buffer is 64K. Which means it can read anything from 1 byte to 64K in an array. You can read a byte stream instead and use `arraysOf` on it to get you arrays of specific size.